evalml/tests/automl_tests/test_automl_search_classification.py

```diff
@@ -717,3 +717,36 @@ def test_tuning_threshold_objective(mock_predict, mock_fit, mock_score, mock_enc
         assert automl.best_pipeline.threshold is None
     else:
         assert automl.best_pipeline.threshold == 0.6
+
+
+@pytest.mark.parametrize("problem_type", ['binary', 'multiclass'])
+@pytest.mark.parametrize("categorical_features", ['none', 'some', 'all'])
+@pytest.mark.parametrize("size", ['small', 'large'])
+@pytest.mark.parametrize("sampling_ratio", [0.8, 0.5, 0.25, 0.2, 0.1, 0.05])
+def test_automl_search_sampler_ratio(sampling_ratio, size, categorical_features, problem_type, mock_imbalanced_data_X_y, has_minimal_dependencies):
+    X, y = mock_imbalanced_data_X_y(problem_type, categorical_features, size)
+    automl = AutoMLSearch(X_train=X, y_train=y, problem_type=problem_type, sampler_method='auto', sampler_balanced_ratio=sampling_ratio)
```

Comment:
For paranoia's sake, can you search with `max_batches=2` and make sure each pipeline gets the right value during the search? Checking the rankings table should be sufficient. 

I see we use the `_frozen_pipelines_parameter` argument in `IterativeAlgorithm` and I don't think we have coverage for that (at least in `test_iterative_algorithm.py`)

Feel free to merge after that!

Reply:
Ah not a bad idea. So you want to run two batches, and assert that all the pipelines were given the right undersampler parameters? That makes sense.

I wonder if this sort of test would fit better in `test_iterative_algorithm.py`, since its testing `_frozen_pipelines_parameter` functionality and isn't specific to the undersampler component, right?

@bchen1116 if you run `search` please don't forget to mock `fit` and `score` :)

Reply:
Yep, adding the test to `test_iterative_algorithm.py`!